doc/Geometry-Validity.md

```diff
@@ -0,0 +1,92 @@
+# Geometry Validity
+
+**TL;DR:**
+
+```ruby
+factory = RGeo::Cartesian.preferred_factory
+bowtie_polygon = factory.polygon(factory.linear_ring([factory.point(0, 4),factory.point(4, 4),factory.point(2, 6),factory.point(0, 0),factory.point(0, 4)]))
+polygon =
+	case bowtie_polygon.invalid_reason
+	when RGeo::ImplHelper::TopologyErrors::SELF_INTERSECTION
```

Comment:
@keithdoggett do we want to simplify the path for this kind of usage? It feels weird to tell a rgeo user to call something under `ImplHelper`

Reply:
Yeah that's smart. Do you think I should put it in the errors module so it's `RGeo::Errors::TopologyErrors` or just make it `RGeo::TopologyErrors` and put it in the top level of the `valid_op.rb` file?

Reply:
I'd go for the first one since every error is under `RGeo::Errors` already, can do it tomorrow